core/trino-main/src/main/java/io/trino/execution/SetTimeZoneTask.java

```diff
@@ -78,7 +78,7 @@ public ListenableFuture<Void> execute(
     {
         String timeZoneId = statement.getTimeZone()
                 .map(timeZone -> getTimeZoneId(timeZone, statement, stateMachine, parameters, warningCollector))
-                .orElse(TimeZone.getDefault().getID());
+                .orElseGet(() -> TimeZone.getDefault().getID());
```

Comment:
(X) I don't think any big logic follows this.

Reply:
reverted due to other reasons (https://github.com/trinodb/trino/pull/15301#discussion_r1040821331)